src/generation.rs

```diff
@@ -2102,6 +2081,40 @@ impl<'a> WasmWrapper<'a> {
         if let Some(from_native) = self.from_native {
             gen_scope.wasm(types, self.ident).push_impl(from_native);
         }
+        if let Some(as_ref) = self.as_ref {
+            gen_scope.wasm(types, self.ident).push_impl(as_ref);
+        }
+    }
+
+    /// native_name is &str since we need to possibly prepend namespacing
+    /// and where we're calling it we'd have to construct a RustType where we
+    /// didn't have to before, but we already had the string.
+    fn add_conversion_methods(&mut self, native_name: &str) {
+        let mut from_wasm = codegen::Impl::new(self.ident.to_string());
+        from_wasm
+            .impl_trait(format!("From<{}>", native_name))
+            .new_fn("from")
+            .arg("native", native_name)
+            .ret("Self")
+            .line("Self(native)");
+            //.line(format!("Self({}(native))", native_name));
+            self.from_wasm = Some(from_wasm);
+        let mut from_native = codegen::Impl::new(native_name);
+        from_native
+            .impl_trait(format!("From<{}>", self.ident.to_string()))
+            .new_fn("from")
+            .arg("wasm", self.ident.to_string())
+            .ret("Self")
+            .line("wasm.0");
+            self.from_native = Some(from_native);
+        let mut as_ref = codegen::Impl::new(self.ident.to_string());
```

Comment:
Do we want `AsMut` too? I feel it's not likely to be useful for us as I think we only ever have a single mutable ref parameter (usually `&mut self`) anyway. I can add `AsMut` here if we want though.

Reply:
No particular preference at the moment

Reply:
I'm leaning more towards not having it then. In the unlikely case we need it it can be easily added in for the appropriate wrapper where it's needed. The `AsRef` one is a lot more generally useful for us for the kind of APIs we write.

Reply:
I updated this to just error. There's a test for the erroring too.